src/model/Patient.ts

```diff
@@ -18,4 +18,5 @@ export default interface Patient extends AbstractDBModel, Name, ContactInformati
   allergies?: Allergy[]
   diagnoses?: Diagnosis[]
   notes?: Note[]
+  index: string
```

Comment:
Could you explain the need for a separate index field other than the id?

Reply:
For paging, the field on which indexing is applied should be unique.
1. Paging with `id` field as index will make the records to appear to be randomly sorted by `fullName`.  
2. Assuming that records should be sorted by `fullName` by default, and `fullName` is non-unique, I thought of combining `fullName` with `patient code` so that uniqueness is maintained.

Reply:
This also fixes #1967

Reply:
@jackcmeyer , can you please provide any feedback on this?